### PR TITLE
step summary item

### PR DESCRIPTION
**Pull request by ds300**
[jira](https://artsyproduct.atlassian.net/browse/PURCHASE-307)

image

### CODE REVIEW

File: src/Styleguide/Components/StepSummaryItem.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+`
+
+export function StepSummaryItem({
+  title,
+  onChange,
+  children,
+  ...others
+}: StepSummaryItemProps) {
+  return (
+    <Wrapper flexDirection="column" width="100%" maxWidth={542} {...others}>
```
Comment (ds300) [inline src/Styleguide/Components/StepSummaryItem.tsx]: Should this maxWidth come from somewhere else? It's not really a breakpoint but it feels like it should be part of the design system somehow.

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit ab322d660b29c1c517ddb02b3850430b1108aa9a

https://deploy-preview-1066--artsy-reaction.netlify.com

File: src/Styleguide/Components/StepSummaryItem.tsx (hunk 1 above)
Comment (zephraph) [reply src/Styleguide/Components/StepSummaryItem.tsx]: Where'd this value come from?

File: src/Styleguide/Components/__stories__/StepSummaryItem.story.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            within 2 business days to handle pickup logistics.
+          </Sans>
+        </StepSummaryItem>
+      </Section>
+      <Section title="Multiple StepSummaryItems">
+        <StepSummaryItem
+          title="Pick up (New York, NY)"
+          onChange={() => console.warn("change")}
+        >
+          <Sans size="2" color="black60" maxWidth={391}>
```
Comment (zephraph) [inline src/Styleguide/Components/__stories__/StepSummaryItem.story.tsx]: This one too?

File: src/Styleguide/Components/StepSummaryItem.tsx (hunk 1 above)
Comment (ds300) [reply src/Styleguide/Components/StepSummaryItem.tsx]: Zeplin

File: src/Styleguide/Components/__stories__/StepSummaryItem.story.tsx (hunk 2 above)
Comment (ds300) [reply src/Styleguide/Components/__stories__/StepSummaryItem.story.tsx]: Also Zeplin

File: src/Styleguide/Components/StepSummaryItem.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import React from "react"
+import styled from "styled-components"
+import { BorderBox } from "Styleguide/Elements/Box"
+import { Flex, FlexProps } from "Styleguide/Elements/Flex"
+import { media } from "Styleguide/Elements/Grid"
+
+interface StepSummaryItemProps extends FlexProps {
+  title: string
+  onChange: () => void
+  children: React.ReactNode | null
```
Comment (alloy) [inline src/Styleguide/Components/StepSummaryItem.tsx]: Just out of curiosity, this only removes the optional typing, right? Why is that needed/helpful?

File: src/Styleguide/Components/StepSummaryItem.tsx (hunk 1 above)
Comment (zephraph) [reply src/Styleguide/Components/StepSummaryItem.tsx]: Ahhh. Be careful with that. Zeplin can be, and often is, wrong. It’s very likely that instead this is intended to align with the grid in some way.

File: src/Styleguide/Components/StepSummaryItem.tsx (hunk 3 above)
Comment (ds300) [reply src/Styleguide/Components/StepSummaryItem.tsx]: `children` isn't in FlexProps, and `StepSummaryItem` is a SFC so doesn't get `children` automatically. If I delete this:

no children

Comment (alloy) [reply src/Styleguide/Components/StepSummaryItem.tsx]: Ah I see. The issue is that the SFC is not typed accordingly. If you were to do this then I think it should work:

```tsx
export const StepSummaryItem: React.SFC<Props> = (...) => {...}
```

Comment (ds300) [reply src/Styleguide/Components/StepSummaryItem.tsx]: Done :+1: This is a neat pattern.

File: src/Styleguide/Components/StepSummaryItem.tsx (hunk 1 above)
Comment (ds300) [reply src/Styleguide/Components/StepSummaryItem.tsx]: Ok for now I've moved the width constraints out of this component, such that the dimensions are controlled by the consumer. I left the widths hard-coded in the story file. They will presumably be constrained by the grid system when in actual use.

File: src/Styleguide/Components/__stories__/StepSummaryItem.story.tsx (hunk 2 above)
Comment (ds300) [reply src/Styleguide/Components/__stories__/StepSummaryItem.story.tsx]: I left this hard-coded in the story file for now. It's not clear to me that this constraint should come from the grid system, because it's not about alignment so much as it is about upholding the legibility of copy that spans multiple lines.

My guess is that it should probably be defined in palette based on the font size? But overridable for situations where having longer lines is the desired behaviour, of course.
